src/pivot-table/data/extract-headers.ts

```diff
@@ -5,17 +5,20 @@ import getKey from "../components/helpers/get-key";
 
 const extractHeaders = (
   hyperCube: ExtendedHyperCube,
-  rowCount: number,
+  visibleTopDimensionInfo: VisibleDimensionInfo[],
   visibleLeftDimensionInfo: VisibleDimensionInfo[],
 ): (null | HeaderCell)[][] => {
-  const matrix: (null | HeaderCell)[][] = Array(visibleLeftDimensionInfo.length)
+  // rowCount cannot be 0, as it will cause an issue when there is no top data but there is left data
+  const rowCount = Math.max(1, visibleTopDimensionInfo.length);
+  const matrix: (null | HeaderCell)[][] = Array(rowCount)
     .fill(null)
-    .map(() => Array.from({ length: rowCount }, () => null));
+    .map(() => Array.from({ length: visibleLeftDimensionInfo.length }, () => null));
```

Comment:
I think you are shifting the matrix here, 
the original matrix is `leftDimInfo.length` * `topDimRowCounts`
but with your changes it's now the `topDimRowCounts` * `leftDimInfo.length`

what i noticed is this file path here: 
1) in use-data.ts file we call [`createHeadersData()`](https://github.com/qlik-oss/sn-pivot-table/blob/df0c53a317fe4fa0fda7e18165692ed82edc1094/src/pivot-table/hooks/use-data.ts#L148) and then pass topDimensionData.rowCount as rowCount 
2) within [`createHeadersData()`](https://github.com/qlik-oss/sn-pivot-table/blob/df0c53a317fe4fa0fda7e18165692ed82edc1094/src/pivot-table/data/headers-data.ts#L11), we forward that as second argument to `extractHeaders(..., rowCount, ...)`
3) and then, in `extractHeaders()` we use that rowCount as [width of matrix at here](https://github.com/qlik-oss/sn-pivot-table/blob/df0c53a317fe4fa0fda7e18165692ed82edc1094/src/pivot-table/data/extract-headers.ts#L13C36-L13C36) and then `visibleLeftDimensionInfo.length` as height of matrix

what happens in here is that you are using `visibleTopDimensionInfo.length` as height and `visibleLeftDimensionInfo.length` as width of the resulted matrix

is that causing any issues? (or may be my brain is broken 🥴 )

Reply:
It correct observartion that I'm shifting the matrix. That is to make it easier when I use it later in the HeaderGrid. I want to iterate each row and for each cell in the row I want to place an element in the grid. I can shift the order of the props so that it makes sense in all function calls

Reply:
as we discussed, the scope of this only falls into header grid and since you are having an eye on it's side effect then it should be fine!